packages/api/src/routes/nfts-upload.js

```diff
@@ -158,16 +157,27 @@ export async function uploadCarWithStat(
     mime_type: mimeType,
     type: uploadType,
     content_cid: stat.rootCid.toV1().toString(),
-    source_cid: stat.rootCid.toString(),
-    dag_size: stat.structure === 'Complete' ? added.bytes : stat.size,
+    source_cid: sourceCid,
+    dag_size: stat.structure === 'Complete' ? stat.blocksSize : stat.size,
```

Comment:
Shouldn't this be only stat.size? We do that in web3 https://github.com/web3-storage/web3.storage/blob/1d101812cbb8db2099dc2fb03447bf2f94f72f40/packages/api/src/car.js#L138 and was expecting same here.

Reply:
Yeah, ok I'm going to revert that `blocksSize` change and always use `stat.size` as in web3.storage.

Turns out using `added.bytes` here was **not** necessarily the size of the DAG (it's just the number of bytes that were imported by cluster). In one of our tests we add _two blobs_ with the _same name_ and the importer creates a DAG with just two blocks (the directory and the second blob) even though the CAR contains 3 blocks (one is orphan).

Resolving this issue will fix the edge case (either by disallowing or allowing it and creating a directory with two files with the same name): https://github.com/ipfs/js-ipfs-unixfs/issues/232